test/test_rake_file_utils.rb

```diff
@@ -287,6 +287,110 @@ def test_sh_show_command
     assert_equal expected_cmd, show_cmd
   end
 
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_echoes_it_fully_long
+    assert_sh_echoes_it_fully(*string_and_output_and_error_long)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_echoes_it_fully_short
+    assert_sh_echoes_it_fully(*string_and_output_and_error_short)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_prints_its_full_output_when_verbose_long
+    verbose true
+    assert_sh_prints_its_full_output(*string_and_output_and_error_long)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_it_raises_a_runtime_error_and_prints_its_full_output_when_verbose_short
+    verbose true
+    assert_sh_prints_its_full_output(*string_and_output_and_error_short)
+  end
+
+  def test_sh_if_a_command_exits_with_error_status_then_providing_a_block_fully_echoes_a_command_when_verbose_long
+    string, output, error_message = string_and_output_and_error_long
+    command = build_command string, output, error_message
+    verbose true
+    actual = capture_subprocess_io do
+      sh(command){|ok, status|}
+    end
+    assert_output_equal actual, command, output, error_message
+  end
+
+  def assert_output_equal(actual, command, output, error_message)
+    # Unix standard-output, standard-error:
+    expected = [output, "#{command}\n#{error_message}"]
+    exp = standard_error_sort expected
+    act = standard_error_sort actual
+    assert_equal exp, act
```

Comment:
Since order is arbitrary why not:

```ruby
assert_match command, error_message
assert_match output, error_message
assert_match actual, error_message
```

This lets us delete a whole method and is less for us to maintain in the future.